napari/_vispy/overlays/scale_bar.py

```diff
@@ -112,13 +116,21 @@ def _on_zoom_change(self, *, force: bool = False):
         # convert desired length to world size
         target_world_pixels = scale_canvas2world * target_canvas_pixels
 
-        # calculate the desired length as well as update the value and units
-        target_world_pixels_rounded, new_dim = self._calculate_best_length(
-            target_world_pixels
-        )
-        target_canvas_pixels_rounded = (
-            target_world_pixels_rounded / scale_canvas2world
-        )
+        # If fixed_width is set, use that value to calculate the scale bar width
+        if self.overlay.fixed_width is not None:
+            target_canvas_pixels_rounded = (
+                self.overlay.fixed_width / scale_canvas2world
+            )
+            new_dim = self.overlay.fixed_width * self._unit.units
+        else:
+            # calculate the desired length as well as update the value and units
+            target_world_pixels_rounded, new_dim = self._calculate_best_length(
+                target_world_pixels
+            )
+            target_canvas_pixels_rounded = (
+                target_world_pixels_rounded / scale_canvas2world
+            )
+
         scale = target_canvas_pixels_rounded
```

Comment:
Nitpick: it's not really `rounded` in both branches (only in the guessed size), so maybe chaning the names to just `target_canvas_pixels` is better.

Reply:
602616c61018eaaa69d5a30b1fb0a1e7262cdbaa